src/pages/_app.tsx

```diff
@@ -11,10 +11,36 @@ import { Sidebar } from '../components/Sidebar/Sidebar';
 import HttpService from '../utils/HttpService';
 import { ROUTES } from '../constants/constants';
 import { useStore } from '../store/store';
+import { useRouter } from 'next/router';
 
-const init = async () => {
+const init = async pathname => {
   const cookbooks = await HttpService.get(ROUTES.COOKBOOKS);
   useStore.setState({ cookbooks });
+
+  const cookbookId = getCookBookIdIfExists(pathname);
+
+  if (cookbookId) {
+    const cookbook = await HttpService.get(ROUTES.COOKBOOK(cookbookId));
```

Comment:
You dont need to do this, we are already getting all of the cookbooks so you can just find the cookbook in the array by ID and set cookbook to that.

Reply:
I think we might already do this in the [:cookbook] route as well

Reply:
aye, we do it in the cookbook route, but that doesn't help if you reload the application while on a section, cause the :cookbook page is never loaded, if you then click a new section the cookbook ID it needs to find the section is never loaded

Reply:
but i agree you dont need the get request :P

Reply:
I think we can make a parent component / wrapper here to handle this instead of doing it in `_app`. Ill read up on Next but this should be possible since sections is a child route of cookbook

Reply:
I'd personally say that cookbooks are the core of your application and the main entity right? I would always load these from `_app` or whatever works as a root

Reply:
^ For cookbooks yes but thats different than the *selected* cookbook